src/components/NcRichContenteditable/NcRichContenteditable.vue

```diff
@@ -471,7 +474,8 @@ export default {
 			if (this.isEmptyValue || !this.maxlength) {
 				return false
 			}
-			return stringLength(this.localValue) > this.maxlength
+			const length = [...this.segmenter.segment(this.localValue)].length
```

Comment:
Ahhh, I love JavaScripts horrific iterator abstraction ...

Reply:
Looks clean to me :see_no_evil:

Reply:
I also would like if they returned an proper array like but on the otherside browser implementation could be probably faster with a generator like it is now.

Reply:
Yeah, it makes sense. I was referring to the incomplete API, for example, a `<iterable>.count()` function like in Rust etc. so that we don't have to copy memory all the time when working with iterators.

Reply:
PS: But it will get better very soon, now that [an important proposal](https://github.com/tc39/proposal-iterator-helpers) was merged.